subsys/testsuite/ztest/Kconfig

```diff
@@ -181,6 +181,58 @@ config TEST_LOGGING_FLUSH_AFTER_TEST
 	default y
 	depends on MULTITHREADING
 
+config ZTEST_MAIN
+	bool "Run ZTest test suites from main"
+	default y
+	help
+	  When disabled, ZTest test suites are not automatically run from main().
+
+config ZTEST_SHELL
+	bool "ZTest shell command"
+	select SHELL
+	select GETOPT
+	select FNMATCH
+	select GETOPT_LONG
+	select SHELL_GETOPT
+	# currently this is broken due to getopt_state / conflicting native definitions on POSIX arch
+	depends on !ARCH_POSIX
+	help
+	  Provide a shell command to run ZTests from the Zephyr shell. The arguments of the 'ztest'
+	  command are inspired by Google Test Framework.
+
+if ZTEST_SHELL
+
+config ZTEST_SHELL_PREFIX
+	string "Prefix for ztest shell options"
+	default ""
+	help
+	  This option is for compatibility with Google Test Framework, where each option specific to
```

Comment:
main motivation here seems to be compatibility with gtest instead of providing full shell support for tests, which is very limiting. And then, most of those options insipred by gtest are not implemented, so basically we go back what can be done already as demonstrated in https://github.com/zephyrproject-rtos/zephyr/pull/58374

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/65437#issuecomment-1859212970

Additional features are pretty trivial to implement, but require refactorization.